src/services/attendee/attendee-router.test.ts

```diff
@@ -1,82 +1,239 @@
 import { TESTER } from "../../../testing/testingTools";
 import { Role } from "../auth/auth-models";
 import { StatusCodes } from "http-status-codes";
-import { Database } from "../../database";
+import { SupabaseDB } from "../../supabase";
 import { v4 as uuidv4 } from "uuid";
 import { getCurrentDay } from "../checkin/checkin-utils";
-import { AttendeeCreateValidator } from "./attendee-validators";
-import { z } from "zod";
+
+const otherEvent = uuidv4();
+const dummyUUID = "00000000-0000-0000-0000-000000000000";
+
+afterAll(async () => {
+    console.log("Cleaning up test environment...");
+
+    try {
+        await SupabaseDB.EVENT_ATTENDANCE.delete()
+            .neq("attendee", "NONEXISTENT_VALUE_THAT_WILL_NEVER_EXIST")
+            .throwOnError();
+        await SupabaseDB.ATTENDEE_ATTENDANCE.delete()
+            .neq("userId", dummyUUID)
+            .throwOnError();
+        await SupabaseDB.EVENTS.delete()
+            .neq("eventId", dummyUUID)
+            .throwOnError();
+        await SupabaseDB.ATTENDEES.delete()
+            .neq("userId", dummyUUID)
+            .throwOnError();
+        await SupabaseDB.REGISTRATIONS.delete()
+            .neq("userId", dummyUUID)
+            .throwOnError();
+        await SupabaseDB.ROLES.delete().neq("userId", dummyUUID).throwOnError();
+    } catch (error) {
+        console.log("Cleanup error (expected):", error);
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 100));
+});
+
+async function createTestEvent() {
+    const testEventId = uuidv4();
+
+    await SupabaseDB.EVENTS.insert({
+        eventId: testEventId,
+        name: "Test Event",
+        description: "Description",
+        startTime: new Date().toISOString(),
+        endTime: new Date(Date.now() + 3600000).toISOString(), // +1hr
+        eventType: "SPECIAL", // use one of the allowed enums
+        isVirtual: false,
+        isVisible: true,
+        location: "Test Location",
+        points: 10,
+    }).throwOnError();
+
+    return testEventId;
+}
+
+type RegistrationOverride = {
+    userId?: string;
+    displayName?: string;
+    email?: string;
+    name?: string;
+    university?: string;
+    degree?: string;
+    isInterestedMechMania?: boolean;
+    isInterestedPuzzleBang?: boolean;
+    dietaryRestrictions?: string[];
+    allergies?: string[];
+};
+
+type AttendeeOverride = {
+    userId?: string;
+    displayName?: string;
+    points?: number;
+    favoriteEvents?: string[];
+    isEligibleTshirt?: boolean;
+    isEligibleCap?: boolean;
+    isEligibleTote?: boolean;
+    isEligibleButton?: boolean;
+    hasRedeemedTshirt?: boolean;
+    hasRedeemedCap?: boolean;
+    hasRedeemedTote?: boolean;
+    hasRedeemedButton?: boolean;
+    hasPriorityMon?: boolean;
+    hasPriorityTue?: boolean;
+    hasPriorityWed?: boolean;
+    hasPriorityThu?: boolean;
+    hasPriorityFri?: boolean;
+    hasPrioritySat?: boolean;
+    hasPrioritySun?: boolean;
+    puzzlesCompleted?: string[];
+};
```

Comment:
are these not just types? can they not just be replaced with types from supabase types?